packages/flutter/lib/src/material/checkbox.dart

```diff
@@ -725,43 +737,75 @@ class _CheckboxPainter extends ToggleablePainter {
   }
 
   void _drawCheck(Canvas canvas, Offset origin, double t, Paint paint) {
-    assert(t >= 0.0 && t <= 1.0);
-    // As t goes from 0.0 to 1.0, animate the two check mark strokes from the
-    // short side to the long side.
     final Path path = Path();
-    const Offset start = Offset(_kEdgeSize * 0.15, _kEdgeSize * 0.45);
-    const Offset mid = Offset(_kEdgeSize * 0.4, _kEdgeSize * 0.7);
-    const Offset end = Offset(_kEdgeSize * 0.85, _kEdgeSize * 0.25);
-    if (t < 0.5) {
-      final double strokeT = t * 2.0;
-      final Offset drawMid = Offset.lerp(start, mid, strokeT)!;
-      path.moveTo(origin.dx + start.dx, origin.dy + start.dy);
-      path.lineTo(origin.dx + drawMid.dx, origin.dy + drawMid.dy);
-    } else {
-      final double strokeT = (t - 0.5) * 2.0;
-      final Offset drawEnd = Offset.lerp(mid, end, strokeT)!;
-      path.moveTo(origin.dx + start.dx, origin.dy + start.dy);
-      path.lineTo(origin.dx + mid.dx, origin.dy + mid.dy);
-      path.lineTo(origin.dx + drawEnd.dx, origin.dy + drawEnd.dy);
+    Offset start;
+    Offset mid;
+    Offset end;
```

Comment:
Can these be `final`?

Reply:
Yes, they never get reassigned. Will change.